### PR TITLE
Updated dependencies

### PR DESCRIPTION
**Pull request by vladislavarsenev**
# Updated following dependencies:
- prettier to 3rd ver
- typescript to 5th ver

# Updated tools:
- minimum Node.js version is 18.x
- minimum Yarn version is 4.5.1

### CODE REVIEW

File: .yarn/install-state.gz
Comment (ayusharma) [inline .yarn/install-state.gz:1]: Is this file needed ?

File: package.json [hunk 1]
```diff
@@ -33,29 +33,33 @@
...
+    "typescript": "5.6.3"
   },
   "peerDependencies": {
     "@vue/compiler-sfc": "3.x",
     "prettier": "2.x - 3.x"
   },
+  "engines": {
+    "node": ">18.12"
+  },
+  "packageManager": "yarn@4.5.1",
```
Comment (ayusharma) [inline package.json:62]: The `packageManager` field is in experimental. I would recommend to not to add. https://nodejs.org/docs/latest-v20.x/api/all.html#all_packages_packagemanager

File: src/utils/is-sort-imports-ignored.ts [hunk 2]
```diff
@@ -6,6 +6,6 @@ import { getAllCommentsFromNodes } from './get-all-comments-from-nodes';
 export const isSortImportsIgnored = (nodes: Statement[]) =>
     getAllCommentsFromNodes(nodes).some(
         (comment) =>
-            comment.loc.start.line === 1 &&
+            comment.loc?.start.line === 1 &&
```
Comment (ayusharma) [inline src/utils/is-sort-imports-ignored.ts:9]: May be :wink
```suggestion
           comment.loc && comment.loc.start.line === 1 &&
```

**Review by ayusharma — CHANGES_REQUESTED** [review]
Thanks @vladislavarsenev. Left couple of small suggestions, other than that everything looks good to me 🚀 Thank you ❤️ I would approve after the discussion on open questions in PR.

File: package.json
Comment (byara) [inline package.json:1]: Just a suggestion (which I think may unblock other things too) is to maybe unpin most of these dependencies?
What do you think @ayusharma?
Because right now we are using exact versions of certain dependencies. I was wondering if we could move away from this.

**Review by byara — APPROVED** [review]
I left one comment but it is not a blocker. Thank you so much for taking a look 👍

File: package.json
Comment (vladislavarsenev) [reply package.json:1]: Do you want to use "compatible with version" or "approximately equivalent" modes? What are benefits of these approaches? I'm curious because last 4 or 5 years we used exact versions on all projects and it was more predictable and convenient.

File: package.json (hunk 1 above)
Comment (vladislavarsenev) [reply package.json]: I've deleted "packageManager" from package.json and removed `corepack` from CI since packageManager was mandatory thing for using it. However, I would like to notice the fact that although all these features are experimental, they have been around for 4 years.

I should ask you about `--experimental-vm-modules` which I added for launching tests with the newest prettier version and it was necessary since prettier@3 utilizes dynamic import. I tried quickly transpile code to omit `--experimental-vm-modules`, but failed. Should I to figure out how to do so without `--experimental-vm-modules`?

File: .yarn/install-state.gz
Comment (vladislavarsenev) [reply .yarn/install-state.gz:1]: Nice catch, thank you!

image

**Review by ayusharma — APPROVED** [review]

File: src/utils/is-sort-imports-ignored.ts (hunk 2 above)
Comment (vladislavarsenev) [reply src/utils/is-sort-imports-ignored.ts]: Fixed, but what a benefit to write in this way? It would never equal 1 if comment.loc was undefined.

File: package.json
Comment (byara) [reply package.json:1]: I think ^ (Compatible with version) is something we want to go for 👍

Comment (vladislavarsenev) [reply package.json:1]: Do you know about any limitations of using ^ right now?

Comment (byara) [reply package.json:1]: I think there were some issues when people had different version of the same dependencies on their project? 🤔 Or am I missing something @ayusharma
